pkg/virt-handler/vm.go

```diff
@@ -237,6 +237,25 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 
 	oldStatus := vmi.DeepCopy().Status
 
+	if domain != nil {
+		var interfaces = vmi.DeepCopy().Status.Interfaces
+		var domain_interfaces []api.Interface = domain.Spec.Devices.Interfaces
+		for _, domain_interface := range domain_interfaces {
```

Comment:
Can you pre-fill a map of vmi interfaces based on their name and then iterate through the list of domain interfaces and look them up in the map?

Reply:
Can you add some unit-tests to `vm_test.go`